jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -165,26 +179,35 @@ public synchronized void execute() throws MojoExecutionException {
         // trigger initialization of the generator instance
         this.getGenerator();
 
+        boolean executedGenerator = false;
+
         if (this.classNames != null) {
             for (String className : this.classNames) {
                 this.getLog().info("Generating JSON Schema for <className>" + className + "</className>");
                 generateSchema(className, false);
+                executedGenerator = true;
             }
         }
 
         if (this.packageNames != null) {
             for (String packageName : this.packageNames) {
                 this.getLog().info("Generating JSON Schema for <packageName>" + packageName + "</packageName>");
                 generateSchema(packageName, true);
+                executedGenerator = true;
             }
         }
+
+        if (!executedGenerator && annotations != null && !annotations.isEmpty()) {
+            this.getLog().info("Generating JSON Schema for all annotated classes");
+            generateSchema("**/*", false);
+        }
     }
 
     /**
      * Generate the JSON schema for the given className.
      *
      * @param classOrPackageName The name or glob pattern of the class or package
-     * @param targetPackage whether the given name or glob pattern refers to a package
+     * @param targetPackage      whether the given name or glob pattern refers to a package
```

Comment:
**suggestion (not-blocking)**: Not ideal to make all these whitespace changes as part of this PR.  Would be nice to minimize the changes to what is required. Also in the unit tests.

Reply:
I think I removed all unnecessary white-space changes